app/models/user.rb

```diff
@@ -14,4 +14,32 @@ def sanitize_phone_number
     phone_number.gsub(/-/, '').prepend("+1")
   end
 
+  def personal_weekly_progress(week_id)
```

Comment:
since we're on the `User` model, you probably don't need to name this "personal_weekly_progress" - "weekly_progress" would work just fine 👍